src/lib/seam/devices/use-set-device-name.ts

```diff
@@ -0,0 +1,93 @@
+import type {
+  DevicesGetParams,
+  DevicesUpdateBody,
+  SeamHttpApiError,
+} from '@seamapi/http/connect'
+import type { Device } from '@seamapi/types/connect'
+import {
+  useMutation,
+  type UseMutationResult,
+  useQueryClient,
+} from '@tanstack/react-query'
+
+import { NullSeamClientError, useSeamClient } from 'lib/seam/use-seam-client.js'
+
+export type UseSetDeviceNameParams = never
+
+export type UseSetDeviceNameData = undefined
+
+export type UseSetDeviceNameMutationVariables = Pick<
+  DevicesUpdateBody,
+  'device_id' | 'name'
+>
+
+type MutationError = SeamHttpApiError
+
+export function useSetDeviceName(
```

Comment:
We like to provide the hooks and match the public API by default. Can we call this `useUpdateDevice`?

It can still only support updating the name (for now)

```
export type UseSetDeviceNameMutationVariables = Pick<
  DevicesUpdateBody,
  'device_id' | 'name'
>
```